### PR TITLE
Use headersSent instead of  _header

### PR DESCRIPTION
**Pull request by maritz**
As discussed in #127 this PR now includes the changes that **do** break node v0.8 and has tests.

I did not remove the travis config for v0.8 since I'm not sure that's something that should be in a PR.

### CODE REVIEW

**Comment by maritz** [comment]
Hm, I just noticed that I branched from my other changes. Should I remove those changes from this PR?

**Comment by dougwilson** [comment]
Hi @maritz I can try to cut them out myself during the merge, but if you can remove them, that may be good especially in case something else changes (either in the other PR or between that PR merging and the major release to include this change).

**Comment by maritz** [comment]
@dougwilson I rebased and cut out the extra commits.

**Comment by lamweili** [comment]
For background information regarding `res._header` and `res.headersSent` equivalence:
https://github.com/nodejs/node/blob/93e0bf9abf350637d772bcce14a5d9527b733300/lib/_http_outgoing.js#L741-L746

We can also add test cases to assert the equivalence (if really required).
The code snippet is available @ https://github.com/expressjs/session/pull/908/commits/35880d6509e6ee6484521f746f936f8a56503be5.

File: index.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ function compression (options) {
         return false
       }
 
-      if (!this._header) {
+      if (!this.headersSent) {
```
[the remark this answers is not in the document]
Comment (bjohansebas) [reply index.js]: good catch

**Comment by bjohansebas** [comment]
> We can also add test cases to assert the equivalence (if really required).
The code snippet is available @ https://github.com/expressjs/session/commit/35880d6509e6ee6484521f746f936f8a56503be5.

@lamweili I don't think we need a test for this, this change was also made in https://github.com/pillarjs/finalhandler/pull/15.

**Review by bjohansebas — APPROVED** [review]
LGTM

**Comment by lamweili** [comment]
> > We can also add test cases to assert the equivalence (if really required).
> The code snippet is available @ https://github.com/expressjs/session/commit/35880d6509e6ee6484521f746f936f8a56503be5.
> 
> @lamweili I don't think we need a test for this, this change was also made in https://github.com/pillarjs/finalhandler/pull/15.

All good, was coming from test/code coverage perspective.

**Review by lamweili — COMMENTED** [review]
LGTM

**Pull request merged**